manifests/fpm.pp

```diff
@@ -0,0 +1,59 @@
+##### LICENSE
+
+# Copyright (c) Skyscrapers (iLibris bvba) 2016 - http://skyscrape.rs
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# == Class nginx::fpm
+#
+# This class will install a PHP-FPM proxy
+#
+define nginx::fpm (
+    $server_name,
+    $port = 80,
+    $http_auth = false,
+    $server_aliases = [],
+    $fpm_port = 9000,
+    $ssl_certificate = undef,
+    $ssl_certificate_key = undef,
+    $ssl_dh_param = undef,
+  ){
+
+  if ! defined(Class['nginx']) {
+    fail('You must include the nginx base class before using any nginx defined resources')
+  }
+
+  if($ssl_certificate == undef or $ssl_certificate_key == undef){
+    $ssl = false
+  } else {
+    $ssl = true
+  }
+
+  file {
+    "/etc/nginx/sites-available/${name}.conf":
+      ensure   => file,
+      content  => template('nginx/etc/nginx/sites-available/fpm.conf.erb'),
+      mode     => '0644',
+      owner    => root,
+      group    => root,
+      require  => Class['nginx::install'],
+      notify   => Class['nginx::service'];
```

Comment:
You only need to restart nginx when creating the `sites-enabled` soft-link, right?